File: scripts/add_credits.py
#!/usr/bin/env python3
"""Add test credits to user account"""

import sys
sys.path.insert(0, '/home/aladdin/Documents/Antigravity/INDXR.AI/backend')

import os
os.chdir('/home/aladdin/Documents/Antigravity/INDXR.AI/backend')

from dotenv import load_dotenv
load_dotenv('.env')

from credit_manager import add_credits
from supabase import create_client

# Get Supabase client
url = os.getenv('SUPABASE_URL')
key = os.getenv('SUPABASE_SERVICE_KEY')

if not url or not key:
    print("Error: Environment variables not found")
    sys.exit(1)

supabase = create_client(url, key)

# Find user by email
try:
    # Query users table
    response = supabase.rpc('get_user_by_email', {'email': 'test3@example.com'}).execute()
    
    if not response.data:
        # Try alternative method
        print("Trying alternative query...")
        # We'll use the service key to query auth.users
        # This requires a custom RPC function or direct database access
        print("\nPlease run this SQL in Supabase SQL Editor:")
        print("SELECT id FROM auth.users WHERE email = 'test3@example.com';")
        print("\nThen run:")
        print("SELECT public.add_credits('USER_ID'::uuid, 100, 'Test credits');")
        sys.exit(0)
    
    user_id = response.data[0]['id']
    print(f"Found user: {user_id}")
    
    # Add credits
    result = add_credits(user_id, 100, 'Test credits for Whisper AI')
    print(f"Result: {result}")
    
    if result.get('success'):
        print(f"\n✅ Successfully added 100 credits!")
        print(f"Previous balance: {result.get('previous_balance')}")
        print(f"New balance: {result.get('new_balance')}")
    else:
        print(f"\n❌ Failed: {result.get('error')}")
        
except Exception as e:
    print(f"Error: {e}")
    print("\n📋 Manual SQL commands:")
    print("1. SELECT id FROM auth.users WHERE email = 'test3@example.com';")
    print("2. SELECT public.add_credits('USER_ID'::uuid, 100, 'Test credits');")

File: backend/add-credits.js
const { createClient } = require('@supabase/supabase-js');
const fs = require('fs');

async function run() {
  try {
    const env = fs.readFileSync('.env', 'utf8');
    const supabaseUrl = env.match(/SUPABASE_URL=(.*)/)[1];
    const supabaseKey = env.match(/SUPABASE_SERVICE_KEY=(.*)/)[1];
    const supabase = createClient(supabaseUrl, supabaseKey);

    console.log("Fetching users...");
    const { data: userData, error: userError } = await supabase.auth.admin.listUsers();
    
    if (userError || !userData || !userData.users || userData.users.length === 0) {
        console.log("No users found.");
        process.exit(1);
    }
    
    const userId = userData.users[0].id;
    console.log("Found user ID:", userId);
    
    // Check current credits in user_credits
    const { data: creditData } = await supabase.from('user_credits').select('*').eq('user_id', userId).single();
    console.log("Current credit data:", creditData);
    
    const currentCredits = creditData ? creditData.credits : 0;
    const newCredits = currentCredits + 20;
    
    // Add 20 credits
    const { error: upsertError } = await supabase.from('user_credits').upsert({
        user_id: userId,
        credits: newCredits
    });
    
    if (upsertError) {
        console.error("Failed to add credits:", upsertError);
    } else {
        console.log(`Successfully added 20 credits. New balance: ${newCredits}`);
    }
  } catch (err) {
      console.error(err);
  } finally {
      process.exit(0);
  }
}
run();
